internal/sharedcomponent/sharedcomponent.go

```diff
@@ -85,20 +74,72 @@ func (c *Component[V]) Unwrap() V {
 
 // Start starts the underlying component if it never started before.
 func (c *Component[V]) Start(ctx context.Context, host component.Host) error {
+	if c.hostWrapper == nil {
+		c.hostWrapper = &hostWrapper{
+			host:           host,
+			sources:        make([]componentstatus.StatusReporter, 0),
+			previousEvents: make([]*componentstatus.StatusEvent, 0),
+		}
+	}
+
+	statusReporter, isStatusReporter := host.(componentstatus.StatusReporter)
+	if isStatusReporter {
+		c.hostWrapper.addSource(statusReporter)
+	}
+
 	var err error
 	c.startOnce.Do(func() {
 		// It's important that status for a shared component is reported through its
 		// telemetry settings to keep status in sync and avoid race conditions. This logic duplicates
 		// and takes priority over the automated status reporting that happens in graph, making the
 		// status reporting in graph a no-op.
-		c.telemetry.ReportStatus(component.NewStatusEvent(component.StatusStarting))
-		if err = c.component.Start(ctx, host); err != nil {
-			c.telemetry.ReportStatus(component.NewPermanentErrorEvent(err))
+		if c.hostWrapper != nil {
```

Comment:
It might be simpler to just always create a hostWrapper and then just let it's `ReportStatus` method do nothing as appropriate.

Reply:
We're actually doing that now, so it is safe to remove these extra checks